packages/proxy-fetcher/src/index.ts

```diff
@@ -94,9 +94,10 @@ export const getFetcher = (
         )
 
         const errorMessage = `Failed request to ${response.url}: ${response.status} ${response.statusText}`
-
-        // Don't retry if we shouldn't
-        if ([404, 403].includes(response.status)) {
+        const urlObj = new URL(response.url)
+        const isJsonApiPath = /\/jsonapi(\/|$)/.test(urlObj.pathname)
```

Comment:
Just to be sure that it tests for `/jsonapi/whatever...` instead of also working for `/some/path/with/jsonapi/whatever`:
```suggestion
        const isJsonApiPath = /^\/jsonapi\//.test(urlObj.pathname)
```